portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,34 +31,32 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> names;
-  private ArrayList<String> emails; 
-  private ArrayList<String> comments;
-
-  private ArrayList<String> messages;
- 
-  @Override
-  public void init() {
-    names = new ArrayList<>();
-    emails = new ArrayList<>();
-    comments = new ArrayList<>();
-
-    messages = new ArrayList<>();
-    messages.add("Howdy");
-    messages.add("Hey, how are you?");
-    messages.add("Good morning, friend!");
-  }
-  
-  //Gets information from the server to display to the user. Picks a random message from the 
-  //messages arraylist and sends the response instance to the screen
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String message = messages.get((int) (Math.random() * messages.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println(message);
+
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Post> posts = new ArrayList<>();
+
+    for (Entity entity : results.asIterable()) {
```

Comment:
Moving this logic to a helper method may make this get easier to parse.

Reply:
yes - just added a todo!